core/common/src/main/java/alluxio/security/group/GroupMappingService.java

```diff
@@ -37,27 +37,68 @@
   class Factory {
     private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
+    // TODO(chaomin): maintain a map from SECURITY_GROUP_MAPPING_CLASS name to cachedGroupMapping.
+    // Currently the single global cached GroupMappingService assumes that there is no dynamic
+    // configuration change for {@link Constants#SECURITY_GROUP_MAPPING_CALSS}.
+    private static GroupMappingService sGroupMappingService = null;
```

Comment:
Before this PR, yes, tons of `GroupMappingService` instances were created....
It was not a big deal because there was no cache for GroupMappingService at all. Now we want to keep a global cache so we should cache those two instances.